File: Shop.Web/shop-web/src/app/models/interfaces/banner.ts
import { IImage } from './image';

export interface Banner {
  id: number;
  type: BannerTypes;
  header: string;
  text?: string;
  isActive: boolean;
  images: IImage[];
}

export interface BannerUpdate {
  id: number;
  header: string;
  text?: string;
  isActive: boolean;
  imageIds?: number[];
}

export enum BannerTypes {
  promo = 0,
  adGallery = 1,
  modal = 2,
}

File: Shop.Database/Models/Banner.cs
using Microsoft.EntityFrameworkCore;
using Shop.Common.Enums;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Shop.Database.Models
{
    [Index(nameof(Type), IsUnique = true)]
    public class Banner
    {
        public int Id { get; set; }

        [Required]
        public BannerType Type { get; set; }

        [Required]
        public string Header { get; set; }

        public string Text { get; set; }

        public bool IsActive { get; set; }

        public ICollection<Image> Images { get; set; } = new List<Image>();
    }
}
